keps/sig-instrumentation/647-apiserver-tracing/README.md

```diff
@@ -0,0 +1,226 @@
+    Please describe the details if so.
+  - Will enabling / using the feature result in non-negligible increase
+    of resource usage (CPU, RAM, disk IO, ...) in any components?
+    Things to keep in mind include: additional in-memory state, additional
+    non-trivial computations, excessive access to disks (including increased
+    log volume), significant amount of data sent and/or received over
+    network, etc. Think through this in both small and large cases, again
+    with respect to the [supported limits][].  **The tracing client library has a small, in-memory cache for outgoing spans.**
+* Rollout, Upgrade, and Rollback Planning
+* Dependencies
+  - Does this feature depend on any specific services running in the cluster
+    (e.g., a metrics service)? **Yes.  In the current version of the proposal, users can run the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector) to configure which backend (e.g. jager, zipkin, etc.) they want telemetry sent to.**
+  - How does this feature respond to complete failures of the services on
+    which it depends?  **Traces will stop being exported, and components will store spans in memory until the buffer is full.  After the buffer fills up, spans will be dropped.**
+  - How does this feature respond to degraded performance or high error rates
+    from services on which it depends? **If the bi-directional grpc streaming connection to the collector cannot be established or is broken, the controller retries the connection every 5 minutes (by default).**
+* Monitoring requirements
+  - How can an operator determine if the feature is in use by workloads?  **Operators are generally expected to have access to the trace backend.**
+  - How can an operator determine if the feature is functioning properly?
+  - What are the service level indicators an operator can use to determine the
+    health of the service?  **Error rate of sending traces in the API Server and OpenTelemetry collector.**
+  - What are reasonable service level objectives for the feature?  **Not entirely sure, but I would expect at least 99% of spans to be sent successfully, if not more.**
+* Troubleshooting
+  - What are the known failure modes?  **The API Server is misconfigured, and cannot talk to the collector.  The collector is misconfigured, and can't send traces to the backend.**
+  - How can those be detected via metrics or logs?  Logs from the component or agent based on the failure mode.
+  - What are the mitigations for each of those failure modes?  **None.  You must correctly configure the collector for tracing to work.**
+  - What are the most useful log messages and what logging levels do they require? **All errors are useful, and are logged as errors (no logging levels required). Failure to initialize exporters (in both controller and collector), failures exporting metrics are the most useful.**
```

Comment:
Errors are not logged per request, but per attempt to send to the collector, correct?

Reply:
Yes.  I clarified this.